### PR TITLE
Feature/149

### PR DESCRIPTION
**Pull request by saadmhybris**
#149 

@xak pls review. All the functionality is there. the big question is whether to keep some styles in the side-nav component or just use the nav component for it.

### CODE REVIEW

**Comment by joseegm** [comment]
Test Version for this PR was deployed

Built with commit 4135ea01ef01ddce28b542ff507df2db4abb16d8

https://deploy-preview-165--fundamental.netlify.com

**Comment by xak** [comment]
@saadmhybris I rebased this test it with the new system ... didn't make much sense otherwise. Please pull your branch before doing further work. 

There were some changes to the toolkit that I updated, mainly the type mixin needs a quoted "string" for the values like `fd-type("-1","header")` and the same for the spacing function like`fd-space("xs")` when using the convenience values.

**Comment by saadmhybris** [comment]
@xak 

This is ready for review again. Please check how the states (is-selected, has-child, is-expanded) is working. I remember this being the naming convention that we are supposed to use. However, i could not find a similar example in any of the other components.

**Comment by xak** [comment]
This is good just a few minor ARIA problems. 

The `aria-expanded` and `aria-haspopup` attrs go on the top control element. When the child list is open it should be `aria-expanded=true` and the child list should have `aria-hidden=false`. Those two will always be opposites — control expanded=true, child hidden=false/control expanded=false, child hidden=true. The child element should have an id and the control element matches with `aria-controls`.*

So in this sense the `expanded` state can act as what you have as the `selected` state (blue background). The `aria-selected` state should only be used for the link of the current view.

You can use the `aria-haspopup` attr in place of the `has-child` class.

```
<ul class="fd-side-nav__list">
  ...
  <li class="fd-side-nav__item">
    <a class="fd-side-nav__link" href="#" aria-haspopup="true" aria-expanded="true" aria-controls="XXXX">
        item link
    </a>
    <ul class="fd-side-nav__sublist" aria-hidden="false" id="XXXX">
      <li class="fd-side-nav__subitem">
        <a class="fd-side-nav__sublink" href="#" aria-selected="true">
         Link
        </a>
      </li>
      ...
    </ul>
  </li>
  ...
</ul>
```

* There is an ID util you can use to generate IDs inside a macro. See dropdown component. The forms use this too to match label[for] to the form ids.

```
{% import "./../utils.njk" as utils %}
...
{%- macro dropdown(properties={}, modifier={}, state={}, aria={}) %}
{%- set _id = utils.id() %}
aria-controls="{{ _id }}"
...
{{ menu({ items: properties.items, id: _id }, modifier.menu) }}
{%- endmacro %}
```

File: scss/components/side-nav.scss [hunk 1]
```diff
@@ -0,0 +1,113 @@
...
+        @include fd-type("-1", "body", "bold");
+
+        //STATES *******************************************
+        &.has-child{
+            background-image: url(#{$fd-forms-select-background-image});
+            background-repeat: no-repeat;
+            background-position: calc(100% - #{$fd-forms-select-width--background-image}) center;
+        }
+
+        &.is-expanded{
```
Comment (xak) [inline scss/components/side-nav.scss]: Add `&[aria-expanded="true"]`

File: scss/components/side-nav.scss [hunk 2]
```diff
@@ -0,0 +1,113 @@
...
+        display: block;
+        margin-bottom: 0;
+        padding: fd-space(4);
+        padding-bottom: 0;
+        @include fd-type("-3", "body", "reg", uppercase);
+        color: $fd-side-nav-link-disabled-color;
+    }
+
+    &__list,
+    &__sublist {
```
Comment (xak) [inline scss/components/side-nav.scss]: There is nothing here to hide this. In the examples, the sublist is visible but has `aria-hidden="true"`

```
<ul class="fd-side-nav__sublist" aria-hidden="true">
  <li class="fd-side-nav__subitem"><a class="fd-side-nav__sublink " href="#">
          Link
      </a>
  </li>
  <li class="fd-side-nav__subitem"><a class="fd-side-nav__sublink " href="#">
          Link
      </a>
  </li>
  <li class="fd-side-nav__subitem"><a class="fd-side-nav__sublink " href="#">
          Link
      </a>
  </li>
</ul>
```

File: scss/components/nav.scss [hunk 3]
```diff
@@ -8,29 +8,21 @@
 */
 $block: #{$fd-namespace}-nav;
 .#{$block} {
```
Comment (xak) [inline scss/components/nav.scss]: Why was this file modified?

**Comment by saadmhybris** [comment]
@xak this is ready for review again.

**Comment by saadmhybris** [comment]
This is ready again.

File: scss/components/side-nav.scss [hunk 4]
```diff
@@ -0,0 +1,136 @@
...
+
+    &__sublist{
+        &[aria-hidden="true"] {
+            display: none;
+        }
+    }
+
+    &__link,
+    &__sublink {
+        padding: $fd-side-nav-title-padding;
```
Comment (xak) [inline scss/components/side-nav.scss]: OK, so there is a small thing here that I discussed with GD. For the top and bottom padding, 12px is too big, 8px is too small. So I agreed we can do 10px. The thing is that the math still works when you take 10 (top) + 20 (line height) + 10 (bottom) it adds to 40px which still adheres to the system.

So here, just do `$fd-side-nav-title-padding+2px` for the top and bottom padding. It's a bit of a cheat but I'm fine with it.

File: test/templates/layout.njk [hunk 5]
```diff
@@ -60,7 +60,7 @@ angular_material/1.1.4</label>
 
 
     </header>
-    <main style="background-color: white;">
+    <main class="fd-has-color-background">
```
Comment (xak) [inline test/templates/layout.njk]: Shouldn't this be `fd-has-background-color-background` not `fd-has-color-background`. The latter changes the text color on the page.
